crates/re_viewport/src/viewport.rs

```diff
@@ -783,39 +792,59 @@ impl TabWidget {
         active: bool,
         gamma: f32,
     ) -> Self {
-        // Not all tabs are for tiles (space views) - some are for containers (e.g. a grid of space views).
-        let space_view = if let Some(egui_tiles::Tile::Pane(space_view_id)) = tiles.get(tile_id) {
-            tab_viewer.space_views.get(space_view_id)
-        } else {
-            None
+        let mut user_named = false;
+        let mut text: egui::WidgetText;
+        let icon;
+        let mut item = None;
```

Comment:
Knowing that each of the hybrid match / if / else paths will set everything seems error prone.

Would clean up to encapsulate as
```
struct TabDisplay {
...
}

impl TabDisplay {
  fn ui(...)
}
```

Then each path can return a TabDisplay struct with field assignment at construction and we can get rid of the muts here as well.

Reply:
I opted to go half-way in that direction: a local struct to cleanup that messy match/if.